base/bitarray.jl

```diff
@@ -335,7 +335,7 @@ end
 @inline try_bool_conversion(x::Real) = x == 0 || x == 1 || throw(InexactError())
 @inline unchecked_bool_convert(x::Real) = x == 1
 
-function copy_to_bitarray_chunks!(Bc::Vector{UInt64}, pos_d::Int, C::Array{<:Real}, pos_s::Int, numbits::Int)
+function copy_to_bitarray_chunks!(Bc::Vector{UInt64}, pos_d::Int, C::AbstractArray{<:Real}, pos_s::Int, numbits::Int)
```

Comment:
May as well get rid of the `{<:Real}` and test something like `B[1:3] = Any[true,false,true]` while we're at it.  The `try_bool_conversion` below will catch any non-Real values.